src/daemon/daemon.cpp

```diff
@@ -3701,60 +3588,73 @@ void mp::Daemon::populate_instance_info(VirtualMachine& vm,
     }
 }
 
-void mp::Daemon::run_commands_at_boot_on_instance(const std::string& name, fmt::memory_buffer& warnings)
+bool mp::Daemon::is_bridged(const std::string& instance_name)
+{
+    const auto& spec = vm_instance_specs[instance_name];
+    const auto& br_interface = get_bridged_interface_name();
+    const auto& br_name = config->factory->bridge_name_for(br_interface);
+
+    return std::any_of(spec.extra_interfaces.cbegin(),
+                       spec.extra_interfaces.cend(),
+                       [&br_interface, &br_name](const auto& network) -> bool {
+                           return network.id == br_interface || network.id == br_name;
+                       });
+}
+
+void mp::Daemon::add_bridged_interface(const std::string& instance_name)
 {
-    auto& vm_specs = vm_instance_specs[name];
-    auto& commands = vm_specs.run_at_boot;
+    // These two use operator[] to access elements because this function is called after existence was checked.
+    mp::VMSpecs& spec = vm_instance_specs[instance_name];
+    mp::VirtualMachine::ShPtr instance = operative_instances[instance_name];
 
-    if (!commands.empty())
+    const auto& br_interface = get_bridged_interface_name();
+    const auto& host_nets = config->factory->networks(); // This will throw if not implemented on this backend.
+    const auto& info = std::find_if(host_nets.cbegin(), host_nets.cend(), [br_interface](const auto& i) {
+        return i.id == br_interface;
```

Comment:
Maybe change the code block line 3612 to 3624 to 
```
    if (const auto info = std::find_if(host_nets.cbegin(), host_nets.cend(), [br_interface](const auto& i) {
            return i.id == br_interface;
        }); info == host_nets.cend())
    {
        throw std::runtime_error(fmt::format(invalid_network_template, br_interface, mp::bridged_interface_key));
    }
    else if (info->needs_authorization && !user_authorized_bridges.count(br_interface))
    {
        throw mp::NonAuthorizedBridgeSettingsException("Cannot update instance settings", instance_name, br_interface);
    }
```
Using if initialization to localize the variable info and also changed `const auto& info` to `const auto info` since std::find_if return iterator by value.

Reply:
Right, looks better the way you propose. Changed.